src/System.Text.Json/tests/Serialization/Value.ReadTests.cs

```diff
@@ -199,41 +200,62 @@ public static void RangePass()
         }
 
         [Fact]
-        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Skipped since NETFX has different semantics and bugs with floating point.")]
         public static void RangePassFloatingPoint()
         {
             // Verify overflow\underflow.
-            // On NETFX these throw.
-            Assert.True(float.IsNegativeInfinity(JsonSerializer.Parse<float>(double.MinValue.ToString(CultureInfo.InvariantCulture))));
-            Assert.True(float.IsPositiveInfinity(JsonSerializer.Parse<float>(double.MaxValue.ToString(CultureInfo.InvariantCulture))));
-            Assert.True(float.IsNegativeInfinity(JsonSerializer.Parse<float?>(double.MinValue.ToString(CultureInfo.InvariantCulture)).Value));
-            Assert.True(float.IsPositiveInfinity(JsonSerializer.Parse<float?>(double.MaxValue.ToString(CultureInfo.InvariantCulture)).Value));
-
-            Assert.True(double.IsNegativeInfinity(JsonSerializer.Parse<double>(double.MinValue.ToString(CultureInfo.InvariantCulture) + "0")));
-            Assert.True(double.IsPositiveInfinity(JsonSerializer.Parse<double>(double.MaxValue.ToString(CultureInfo.InvariantCulture) + "0")));
-            Assert.True(double.IsNegativeInfinity(JsonSerializer.Parse<double?>(double.MinValue.ToString(CultureInfo.InvariantCulture) + "0").Value));
-            Assert.True(double.IsPositiveInfinity(JsonSerializer.Parse<double?>(double.MaxValue.ToString(CultureInfo.InvariantCulture) + "0").Value));
+            if (PlatformDetection.IsFullFramework)
```

Comment:
We are following this pattern elsewhere already.

Reply:
Consider if this split would be better served by making a helper

```C#
AssertFloatingPointBehavior(float.NegativeInfinity, float.MinValue);
AssertFloatingPointBehavior((float?)float.NegativeInfinity, float.MinValue);
AssertFloatingPointBehavior(double.NegativeInfinity, double.MinValue);
AssertFloatingPointBehavior((double?)double.NegativeInfinity, double.MinValue);

...

private static void AssertFloatingPointBehavior<T>(T parsedValue, T inputSeed)
{
    string toParse = inputSeed.ToString(CultureInfo.InvariantCulture) + "0";

    if (PlatformDetection.IsFullFramework)
    {
        Assert.Throws<FormatException>(() => JsonSerializer.Parse<T>(toParse));
    }
    else
    {
        Assert.Equal(parsedValue, JsonSerializer.Parse<T>(toParse));
    }
}
```

That way cases are more obviously (because it's enforced) added symmetrically.

Reply:
Fixed.